tests/compatibility-test.py

```diff
@@ -58,6 +59,24 @@ def test_cluster_info(self):
         """Execute "print(ray.cluster_resources())" in the head Pod."""
         EasyJobRule().assert_rule()
 
+    def test_probe_injection(self):
```

Comment:
* Run the test with `kuberay/operator:v1.0.0`. The test should fail.
  ```sh 
  RAY_IMAGE=rayproject/ray:nightly OPERATOR_IMAGE=kuberay/operator:v1.0.0 python3 tests/compatibility-test.py BasicRayTestCase 2>&1
  ```
  <img width="871" alt="Screen Shot 2023-11-27 at 11 18 29 PM" src="https://github.com/ray-project/kuberay/assets/20109646/ec6ea200-a418-4eb2-87d6-b043dfdf699e">

* Run the test with `controller:latest` (this PR). The test should pass.
   ```sh 
  RAY_IMAGE=rayproject/ray:nightly OPERATOR_IMAGE=controller:latest python3 tests/compatibility-test.py BasicRayTestCase 2>&1
  ```
  <img width="1038" alt="Screen Shot 2023-11-27 at 11 19 59 PM" src="https://github.com/ray-project/kuberay/assets/20109646/fcfc4a8a-8b04-4926-8612-3d808ea60042">